android/src/main/java/io/mosip/tuvali/ble/central/state/StateHandler.kt

```diff
@@ -256,8 +257,14 @@ class StateHandler(
     try {
       super.dispatchMessage(msg)
     } catch (e: Throwable) {
-      listener.onException(StateHandlerException("Exception in Central State Handler", e))
-      Log.d(logTag, "dispatchMessage " + e.message)
+      var bleException: BLEException = WalletStateHandlerException("Exception in Central State Handler", e);
+
+      if(e is BLEException) {
+        bleException = e;
+      }
+
+      listener.onException(bleException)
```

Comment:
Above code is not required IMO.

it can be as simple as this `listener.onException(WalletStateHandlerException("Exception in Central State Handler", e))`